governance/pyth_staking_sdk/src/pyth-staking-client.ts

```diff
@@ -680,19 +681,53 @@ export class PythStakingClient {
       stakeAccountPositions,
     );
     const allPublishers = extractPublisherData(poolData);
-    const publishers = allPublishers.filter(({ pubkey }) =>
-      stakeAccountPositionsData.data.positions.some(
-        ({ targetWithParameters }) =>
-          targetWithParameters.integrityPool?.publisher.equals(pubkey),
-      ),
-    );
+    const publishers = allPublishers
+      .map((publisher) => {
+        const positionsWithPublisher =
+          stakeAccountPositionsData.data.positions.filter(
+            ({ targetWithParameters }) =>
+              targetWithParameters.integrityPool?.publisher.equals(
+                publisher.pubkey,
+              ),
+          );
+
+        let lowestEpoch;
+        for (const position of positionsWithPublisher) {
+          if (
+            lowestEpoch === undefined ||
+            position.activationEpoch < lowestEpoch
+          ) {
+            lowestEpoch = position.activationEpoch;
+          }
```

Comment:
can we use bigintMax here?

Reply:
I guess we have to use bigintMin. It might be worth making it a funciton since it'll be used in 2 different places.